src/DefaultBuilder/test/Microsoft.AspNetCore.Tests/WebApplicationTests.cs

```diff
@@ -2697,7 +2699,7 @@ public void DebugView_UseMiddleware_HasMiddleware()
         // 3. Generated delegate name from app.Use(...)
         Assert.Collection(debugView.Middleware,
             m => Assert.Equal(typeof(MiddlewareWithInterface).FullName, m),
-            m => Assert.Equal("Microsoft.AspNetCore.Authentication.AuthenticationMiddleware", m),
+            m => Assert.StartsWith("Microsoft.AspNetCore.Builder.AuthAppBuilderExtensions", m),
```

Comment:
Not sure exactly what you mean. I updated the check because I wanted to avoid having the generated name of the delegate-based middleware here. I think it stays mostly consistent but I wanted to avoid typing out the whole thing.

Also, it's a helpful indicator of what codepath you went through (re-routed middleware or plain middleware).

Reply:
I meant in the case of consumable data, can customers only see this in the debugger? Or should we treat it more like public API and try to have nicer names show up?